mayham-game/Goal.gd

```diff
@@ -4,11 +4,48 @@ extends Area2D
 # var a = 2
 # var b = "textvar"
 
+var player_list = preload("res://PlayerLinkedList.gd")
+
+var players_in_goal = null
+var players_not_in_goal = null
+
+func init(players):
+	players_in_goal = player_list.new()
+	players_not_in_goal = player_list.new()
+	for player in players:
+		players_not_in_goal.push_back(player)
+
 func _ready():
+	connect("body_entered", self, "_on_Goal_entered")
+	connect("body_exited", self, "_on_Goal_exited")
+
 	# Called when the node is added to the scene for the first time.
 	# Initialization here
 	pass
 
+func _on_Goal_entered(body):
+	if body.is_class("StaticBody2D"):
+		return
+	var player = players_not_in_goal.pop_by_number(body.get_number())
+	players_in_goal.push_back(player)
+
+func _on_Goal_exited(body):
```

Comment:
We'll want to check that the body is a player here as well, mostly for consistency but also for when the goal starts moving around

Reply:
I was attempting to find a way to check that the type was of a player, but I couldn't find any. So I settled for check if the body is not of type StaticBody2D. Do you feel like it would be sufficient to check if body type is KinematicBody2d?

Reply:
I just changed the code to check the type for KinematicBody2D. I will be committing it shortly.

Reply:
Looks good